### PR TITLE
[SDEV3-934]: Add convenience methods for modifying button props

### PR DESCRIPTION
**Pull request by brandonminch**
## What does this PR do?

Adds convenience methods for modifying button props on confirm modals

## Type

- [X] Feature
- [ ] Bug
- [ ] Tech debt

## What are the relevant tickets?

- [SDEV3-934](https://sprucelabsai.atlassian.net/browse/SDEV3-934)

## What gif best describes this PR or how it makes you feel?

buttons

### CODE REVIEW

File: packages/spruce-skill/interface/pages/skill-views/dashboard_location/index.js [hunk 1]
```diff
@@ -111,25 +111,25 @@ class DashboardLocationPage extends React.Component {
...
 			}
-		)
+		})
 	}
 
 	handleShowConfirmationWithInputModal = () => {
-		this.confirm.show(
-			{
+		this.confirm.show({
+			data: {
```
Comment (liquidg3) [inline packages/spruce-skill/interface/pages/skill-views/dashboard_location/index.js]: everywhere else i see us putting callbacks (`onConfirm` and `onCancel` on the same level as all other props. Check screen.

https://www.dropbox.com/s/tknfd6wofzrsfkh/Screenshot%202019-06-08%2012.01.03.png?dl=0

So i think we should drop `data` here and do it all on 1 level.

File: packages/spruce-next-helpers/src/skillskit/index.js [hunk 2]
```diff
@@ -263,7 +263,15 @@ const skill = {
 
 	confirm: function() {
 		const confirm = {
-			show: (data: Object, onConfirm?: Function, onCancel?: Function) => {
+			show: ({
+				data,
```
Comment (liquidg3) [inline packages/spruce-next-helpers/src/skillskit/index.js]: See comment below, but we usually mixin callbacks with other props, like this screen: https://www.dropbox.com/s/ruzcb29dq6scstl/Screenshot%202019-06-08%2012.02.27.png?dl=0

so i think we should drop `data` from here.

File: packages/spruce-skill/interface/pages/skill-views/dashboard_location/index.js (hunk 1 above)
Comment (brandonminch) [reply packages/spruce-skill/interface/pages/skill-views/dashboard_location/index.js]: updated

File: packages/spruce-next-helpers/src/skillskit/index.js (hunk 2 above)
Comment (brandonminch) [reply packages/spruce-next-helpers/src/skillskit/index.js]: updated

**Pull request merged**